ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/question_answering/SentenceHighlightingQATranslator.java

```diff
@@ -0,0 +1,244 @@
+
+            NDArray tokenTypeIdsArray = manager.create(encodings.getTypeIds());
+            tokenTypeIdsArray.setName(TOKEN_TYPE_IDS);
+
+            return new NDList(indicesArray, attentionMaskArray, tokenTypeIdsArray);
+        } catch (Exception e) {
+            throw new RuntimeException(String.format(Locale.ROOT, "Error processing input: %s", e.getMessage()), e);
+        }
+    }
+
+    @Override
+    public Output processOutput(TranslatorContext ctx, NDList list) {
+        try {
+            Output output = new Output(200, "OK");
+            List<ModelTensor> outputs = new ArrayList<>();
+
+            @SuppressWarnings("unchecked")
+            List<Sentence> sentences = (List<Sentence>) ctx.getAttachment(KEY_SENTENCES);
+            String question = (String) ctx.getAttachment(KEY_QUESTION);
+            if (sentences == null || sentences.isEmpty()) {
+                return createErrorOutput("No valid sentences found in the provided context");
+            }
+
+            boolean[] isRelevant = new boolean[sentences.size()];
+
+            // Check if we have valid output from the model
+            if (list == null || list.isEmpty()) {
+                return createErrorOutput("Model returned empty or null output");
+            }
+
+            // The model returns a tensor where 0 means relevant, 1 means not relevant
+            NDArray binaryPreds = list.getFirst();
+
+            // Validate prediction shape
+            if (binaryPreds.getShape().dimension() == 0 || binaryPreds.getShape().get(0) == 0) {
+                return createErrorOutput(String.format("Invalid prediction shape: %s", binaryPreds.getShape()));
+            }
+
+            // Convert to boolean array
+            for (int i = 0; i < Math.min(sentences.size(), binaryPreds.getShape().get(0)); i++) {
+                try {
+                    long predValue = binaryPreds.getLong(i);
+                    isRelevant[i] = isRelevantPrediction(predValue);
+                } catch (Exception e) {
+                    log.warn(String.format("Error processing prediction for sentence %d: %s", i, e.getMessage()));
+                    isRelevant[i] = false;
+                }
+            }
+
+            // Create sentence data objects
+            List<SentenceData> sentenceDataList = new ArrayList<>();
+            for (int i = 0; i < sentences.size(); i++) {
+                Sentence sentence = sentences.get(i);
+                boolean relevant = isRelevant[i];
+                sentenceDataList.add(new SentenceData(sentence.getText(), relevant, sentence.getPosition()));
+            }
+
+            // Prepare output list for relevant sentences
+            List<Map<String, Object>> relevantSentenceDetails = getRelevantSentenceDetails(sentenceDataList, sentences);
+
+            // Create output tensor with a single "highlights" field
+            outputs.add(new ModelTensor(FIELD_HIGHLIGHTS, gson.toJson(relevantSentenceDetails)));
```

Comment:
same here, you can use StringUtil.toJson, then we can manage the helpful function in the same package